daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
I think there is a misunderstanding here. 
Because of the choice of passing an explicit cli flag (`internal`) as part of a map indexed by libnetwork key (from libnetwork.netlabel) and called `NetworkOptions`.

This is a protocol between docker client and docker daemon on how to carry cli options in the network create body message. _It has nothing to do with the libnetwork API_.

If we choose to carry these flags via key value pairs, then @calavera's suggestion is perfectly fine, it will have no impact on the libnetwork API calls. But still, why use `netlabel.internal` to index it, it has nothing to do with libnetwork).

But I myself think it would be much better to have an explicit type for each well-known cli option, rather than trying to fit it in a map of strings. Proposed solution works well for true/false flags, but it is not suitable for more complex structure (see the ipam related configs) which would now need to be encoded/decoded to a string form.
Also why do we want to re-do the job json encoding/decoding already provides.

I am not aware if it was called to not add new types to the `NetworkCreate` structure.
If this is not the case, I am strongly in favor of a new type: `NetworkCreate.Internal`.
The `NetworkCreate.options` we have is in fact one type for the driver options.

Also, adding new types does not break backward compatibility.